packages/block-editor/src/components/link-control/index.js

```diff
@@ -241,8 +298,7 @@ function LinkControl( {
 	};
 
 	const resetInternalValues = () => {
-		setInternalUrlInputValue( value?.url );
-		setInternalTextInputValue( value?.title );
+		setInternalControlValue( value );
```

Comment:
Instead of resetting individual properties, "reset" now just means "sync back to whatever `value` was/is"